legiti/src/main/java/com/legiti/helpers/LegitiDeviceData.kt

```diff
@@ -81,14 +92,40 @@ internal class LegitiDeviceData(private val androidContext: Context) {
         return networkList.contains("tun0") || networkList.contains("ppp0")
     }
 
+    private fun ByteArray.toHexString() = joinToString("") { "%02x".format(it) }
+
+    // Based on https://beltran.work/blog/2018-03-27-device-unique-id-android/
+    // and https://stackoverflow.com/a/59050182
     private fun getDeviceFingerprint(): String? {
-        val device_fingerprint = Settings.Secure.getString(this.androidContext.contentResolver, Settings.Secure.ANDROID_ID) ?: return null
+        var meDrm: MediaDrm? = null
+        var deviceFingerprint: String? = null
+
+        for (provider in this.COMMON_DRM_PROVIDERS) {
+            try {
+                meDrm = MediaDrm(provider)
+                var rawFingerprint = meDrm.getPropertyByteArray(MediaDrm.PROPERTY_DEVICE_UNIQUE_ID)
+                var md = MessageDigest.getInstance("SHA-256")
```

Comment:
we're still hashing it?

Reply:
Yes, but the thing we are hashing is always the same. We are hashing so it gets a fixed-length